core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -259,13 +274,15 @@ public void retryWithH1C() {
     }
 
     @Override
-    public boolean isActive() {
-        return active;
+    public boolean canAcquire() {
+        return active && responseDecoder != null && !responseDecoder.needsToDisconnectWhenFinished();
```

Comment:
Took another look and I realized that `responseDecoder` is guaranteed to be set when a channel is added to the pool.
I wonder if it's just worth doing an assertion so that there is less confusion:

```
assert responseDecoder != null
return active && !responseDecoder.needsToDisconnectWhenFinished();
```